test/commands/generate/fromTemplate.test.ts

```diff
@@ -9,49 +9,60 @@ const generalOptions = [
   '@asyncapi/minimaltemplate',
 ];
 
-async function cleanup(filepath: string) {
+function cleanup(filepath: string) {
   rimraf.sync(filepath);
 }
 
 describe('template', () => {
+  afterAll(() => {
+    cleanup('./test/docs');
+  });
   test
     .stdout()
-    .command([...generalOptions, '--output=./test/docs', '--force-write'])
+    .command([...generalOptions, '--output=./test/docs/1', '--force-write'])
     .it('should generate minimal tempalte', (ctx, done) => {
       expect(ctx.stdout).toContain(
-        'Check out your shiny new generated files at ./test/docs.\n\n'
+        'Check out your shiny new generated files at ./test/docs/1.\n\n'
       );
-      cleanup('./test/docs');
+      cleanup('./test/docs/1');
       done();
     });
 
-  test
-    .stderr()
-    .command([...generalOptions, '--output=./test/doc'])
-    .it(
-      'should throw error if output folder is in a git repository',
-      (ctx, done) => {
-        expect(ctx.stderr).toContain(
-          'Error: "./test/doc" is in a git repository with unstaged changes.'
-        );
-        cleanup('./test/doc');
-        done();
-      }
-    );
+  describe('git clash', () => {
+    const pathToOutput = './test/docs/2';
+    beforeAll(() => {
+      fs.mkdirSync(pathToOutput, { recursive: true });
+      // Write a random file to trigger that dir has unstaged changes.
+      fs.writeFileSync(path.join(pathToOutput, 'random.md'), '');
+    });
+    test
+      .stderr()
+      .command([...generalOptions, '--output=./test/docs/2'])
+      .it(
+        'should throw error if output folder is in a git repository',
+        (ctx, done) => {
+          expect(ctx.stderr).toContain(
+            'Error: "./test/docs/2" is in a git repository with unstaged changes.'
+          );
+          cleanup('./test/docs/2');
+          done();
+        }
+      );
+  });
```

Comment:
I dont quite understand why this worked before, but I guess that's just race conditions in a nutshell 🤷 

I refactored the tests so each tests are completely siloed from the other. Further, this test is specifically set up to trigger the error where before, I have no idea how it was triggered 😆